src/main/java/org/opensearch/security/transport/SecurityInterceptor.java

```diff
@@ -226,12 +227,12 @@ && getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_ROL
                 );
             }
 
-            if (useJDKSerialization) {
+            if (!useJDKSerialization) {
                 Map<String, String> jdkSerializedHeaders = new HashMap<>();
                 HeaderHelper.getAllSerializedHeaderNames()
                     .stream()
                     .filter(k -> headerMap.get(k) != null)
-                    .forEach(k -> jdkSerializedHeaders.put(k, Base64Helper.ensureJDKSerialized(headerMap.get(k))));
+                    .forEach(k -> jdkSerializedHeaders.put(k, Base64Helper.ensureCustomSerialized(headerMap.get(k))));
```

Comment:
if the version is 2.11.0, why it would not be custom serialized?

Reply:
We by default will serialize all headers via JDK and if during send we evaluate that the remote node understands custom, we invoke `ensureCustomSerialized` on the intermediate header map.

2.11.0 will receive custom serialized headers only.